In [1]:
#import libaries

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [3]:
#import csv file

In [4]:
data = pd.read_csv('International_Breweries.csv')

In [5]:
data.head()

SALES_ID SALES_REP               EMAILS       BRANDS  PLANT_COST  \
0     10101   Jardine       jard@gmail.com       trophy         150   
1     10102      Gill      gillhell@uk.com    budweiser         250   
2     10103   Sorvino  sorvi2000@gmail.com  castle lite         180   
3     10104     Jones    jone.ai@yahoo.com  eagle lager         170   
4     10105   Andrews       andy@gmail.com         hero         150   

   UNIT_PRICE  QUANTITY    COST  PROFIT COUNTRIES     REGION     MONTHS  YEARS  
0         200       725  145000   36250     Ghana   Southeast   January   2019  
1         500       815  407500  203750   Nigeria        west  February   2018  
2         450       937  421650  252990      Togo  southsouth     March   2018  
3         250       765  191250   61200     Benin   northwest     April   2018  
4         200       836  167200   41800   Senegal  northeast        May   2017

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1047 entries, 0 to 1046
Data columns (total 13 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   SALES_ID    1047 non-null   int64 
 1   SALES_REP   1047 non-null   object
 2   EMAILS      1047 non-null   object
 3   BRANDS      1047 non-null   object
 4   PLANT_COST  1047 non-null   int64 
 5   UNIT_PRICE  1047 non-null   int64 
 6   QUANTITY    1047 non-null   int64 
 7   COST        1047 non-null   int64 
 8   PROFIT      1047 non-null   int64 
 9   COUNTRIES   1047 non-null   object
 10  REGION      1047 non-null   object
 11  MONTHS      1047 non-null   object
 12  YEARS       1047 non-null   int64 
dtypes: int64(7), object(6)
memory usage: 81.9+ KB


In [7]:
data.isnull().sum()

SALES_ID      0
SALES_REP     0
EMAILS        0
BRANDS        0
PLANT_COST    0
UNIT_PRICE    0
QUANTITY      0
COST          0
PROFIT        0
COUNTRIES     0
REGION        0
MONTHS        0
YEARS         0
dtype: int64

In [8]:
data.describe()

SALES_ID   PLANT_COST   UNIT_PRICE     QUANTITY           COST  \
count   1047.000000  1047.000000  1047.000000  1047.000000    1047.000000   
mean   10624.000000   152.989494   271.728749   850.737345  231034.049666   
std      302.387169    53.125477   133.378553    86.991652  115658.626612   
min    10101.000000    80.000000   150.000000   700.000000  105300.000000   
25%    10362.500000    90.000000   150.000000   776.000000  141425.000000   
50%    10624.000000   150.000000   200.000000   853.000000  184400.000000   
75%    10885.500000   180.000000   450.000000   927.000000  351450.000000   
max    11147.000000   250.000000   500.000000  1000.000000  499500.000000   

              PROFIT        YEARS  
count    1047.000000  1047.000000  
mean   100847.583572  2017.941738  
std     77581.775930     0.808325  
min     35000.000000  2017.000000  
25%     46880.000000  2017.000000  
50%     58080.000000  2018.000000  
75%    191205.000000  2019.000000  
max    269190.000000  2019.000000

In [9]:
#/*PROFIT ANALYSIS*/
#1. /*Within the space of the last three years, what was the profit worth of the breweries,
#inclusive of the anglophone and the francophone territories

In [10]:
data['PROFIT'].sum()

105587420

The total profit made by the breweries in 2017, 2018 and 2019 is 105,587,420 

In [11]:
#2./*Compare the total profit between these two territories in order for the territory manager, 
#Mr. Stone make strategic decision that will aid profit maximization in 2020*/

In [12]:
data[['COUNTRIES','PROFIT']]

COUNTRIES  PROFIT
0        Ghana   36250
1      Nigeria  203750
2         Togo  252990
3        Benin   61200
4      Senegal   41800
...        ...     ...
1042      Togo   57720
1043     Benin   44600
1044   Senegal  204000
1045     Ghana  253530
1046   Nigeria   74800

[1047 rows x 2 columns]

In [13]:
#create a function for anglophone and francophone countries

In [14]:
def check_anglophone(country):
    if (country == 'Nigeria') or (country =='Ghana'):
        return True
    else:
        return False
    
    
def check_francophone(country):
    if (country =='Senegal') or (country =='Togo') or (country =='Benin'):
        return True
    else:
        return False

In [15]:
data[data['COUNTRIES'].apply(check_francophone)]

SALES_ID SALES_REP               EMAILS       BRANDS  PLANT_COST  \
2        10103   Sorvino  sorvi2000@gmail.com  castle lite         180   
3        10104     Jones    jone.ai@yahoo.com  eagle lager         170   
4        10105   Andrews       andy@gmail.com         hero         150   
7        10108     Jones    jone.ai@yahoo.com       trophy         150   
8        10109    Morgan   morganny@gmail.com    budweiser         250   
...        ...       ...                  ...          ...         ...   
1038     11139      Gill      gillhell@uk.com  castle lite         180   
1039     11140   Sorvino  sorvi2000@gmail.com  eagle lager         170   
1042     11143   Jardine       jard@gmail.com   grand malt          90   
1043     11144   Jardine       jard@gmail.com       trophy         150   
1044     11145   Andrews       andy@gmail.com    budweiser         250   

      UNIT_PRICE  QUANTITY    COST  PROFIT COUNTRIES        REGION   \
2            450       937  421650  252990      Togo     southsouth   
3            250       765  191250   61200     Benin      northwest   
4            200       836  167200   41800   Senegal     northeast    
7            200       812  162400   40600      Togo           west   
8            500       700  350000  175000     Benin     southsouth   
...          ...       ...     ...     ...       ...            ...   
1038         450       907  408150  244890     Benin      Southeast   
1039         250       949  237250   75920   Senegal           west   
1042         150       962  144300   57720      Togo     northeast    
1043         200       892  178400   44600     Benin  northcentral    
1044         500       816  408000  204000   Senegal      Southeast   

         MONTHS  YEARS  
2         March   2018  
3         April   2018  
4           May   2017  
7        August   2018  
8     September   2019  
...         ...    ...  
1038       July   2017  
1039     August   2017  
1042   November   2017  
1043   December   2017  
1044    January   2019  

[627 rows x 13 columns]

In [16]:
df = data[data['COUNTRIES'].apply(check_francophone)]
df.groupby(['COUNTRIES'])['PROFIT'].sum()

COUNTRIES
Benin      20961970
Senegal    21485190
Togo       20751000
Name: PROFIT, dtype: int64

In [17]:
francophone = df.groupby(['COUNTRIES'])['PROFIT'].sum().loc[['Benin','Senegal', 'Togo']].sum()
francophone

63198160

In [18]:
#profits made in anglophone countries

In [19]:
data[data['COUNTRIES'].apply(check_anglophone)]

SALES_ID SALES_REP              EMAILS       BRANDS  PLANT_COST  \
0        10101   Jardine      jard@gmail.com       trophy         150   
1        10102      Gill     gillhell@uk.com    budweiser         250   
5        10106   Jardine      jard@gmail.com    beta malt          80   
6        10107  Thompson        thomp@uk.com   grand malt          90   
10       10111    Parent     parentty@uk.com  eagle lager         170   
...        ...       ...                 ...          ...         ...   
1036     11137    Kivell  kivel_go@yahoo.com       trophy         150   
1040     11141  Thompson        thomp@uk.com         hero         150   
1041     11142   Andrews      andy@gmail.com    beta malt          80   
1045     11146     Jones   jone.ai@yahoo.com  castle lite         180   
1046     11147    Morgan  morganny@gmail.com  eagle lager         170   

      UNIT_PRICE  QUANTITY    COST  PROFIT COUNTRIES        REGION   \
0            200       725  145000   36250     Ghana      Southeast   
1            500       815  407500  203750   Nigeria           west   
5            150       798  119700   55860     Ghana  northcentral    
6            150       954  143100   57240   Nigeria      Southeast   
10           250       861  215250   68880     Ghana     northeast    
...          ...       ...     ...     ...       ...            ...   
1036         200       974  194800   48700   Nigeria     northeast    
1040         200       903  180600   45150     Ghana     southsouth   
1041         150       740  111000   51800   Nigeria      northwest   
1045         450       939  422550  253530     Ghana           west   
1046         250       935  233750   74800   Nigeria     southsouth   

         MONTHS  YEARS  
0       January   2019  
1      February   2018  
5          June   2019  
6          July   2017  
10     November   2017  
...         ...    ...  
1036        May   2017  
1040  September   2019  
1041    October   2018  
1045   February   2017  
1046      March   2019  

[420 rows x 13 columns]

In [20]:
df1 = data[data['COUNTRIES'].apply(check_anglophone)]
df1.groupby(['COUNTRIES'])['PROFIT'].sum()

COUNTRIES
Ghana      21013050
Nigeria    21376210
Name: PROFIT, dtype: int64

In [21]:
anglophone = df1.groupby(['COUNTRIES'])['PROFIT'].sum().loc[['Ghana','Nigeria',]].sum()

print('Total profit made in the Anglophone region: ', anglophone)
print('Total profit made in the Francophone region: ', francophone)

Total profit made in the Anglophone region:  42389260
Total profit made in the Francophone region:  63198160


In [22]:
#/*Country that generated the highest profit in 2019*/

In [23]:
yr_2019 = data[data['YEARS'] == 2019]
yr_2019.head(3)

SALES_ID SALES_REP              EMAILS     BRANDS  PLANT_COST  UNIT_PRICE  \
0     10101   Jardine      jard@gmail.com     trophy         150         200   
5     10106   Jardine      jard@gmail.com  beta malt          80         150   
8     10109    Morgan  morganny@gmail.com  budweiser         250         500   

   QUANTITY    COST  PROFIT COUNTRIES        REGION      MONTHS  YEARS  
0       725  145000   36250     Ghana      Southeast    January   2019  
5       798  119700   55860     Ghana  northcentral        June   2019  
8       700  350000  175000     Benin     southsouth  September   2019

In [24]:
yr_2019.groupby(['COUNTRIES'])['PROFIT'].sum().sort_values(ascending=False).head(1)


COUNTRIES
Ghana    7144070
Name: PROFIT, dtype: int64

The Country with the most profit in the year 2019 is Ghana

In [25]:
#/*Help him find the year with the highest profit.*/

In [26]:
data.groupby(['YEARS'])['PROFIT'].sum().sort_values(ascending = False).head(1)

YEARS
2017    38503320
Name: PROFIT, dtype: int64

The most profitable year is 2017

In [27]:
# /*Which month in the three years were the least profit generated?*/

In [28]:
data.groupby(['MONTHS'])['PROFIT'].sum().sort_values(ascending = True).head(1)

MONTHS
April    8573830
Name: PROFIT, dtype: int64

The least profitable month recorded in the 3years was April

In [29]:
# /*Compare the profit in percentage for each of the month in 2019*/

In [30]:
yr_2019.head(5)

SALES_ID SALES_REP              EMAILS     BRANDS  PLANT_COST  UNIT_PRICE  \
0      10101   Jardine      jard@gmail.com     trophy         150         200   
5      10106   Jardine      jard@gmail.com  beta malt          80         150   
8      10109    Morgan  morganny@gmail.com  budweiser         250         500   
11     10112     Jones   jone.ai@yahoo.com       hero         150         200   
15     10116     Jones   jone.ai@yahoo.com  budweiser         250         500   

    QUANTITY    COST  PROFIT COUNTRIES        REGION      MONTHS  YEARS  
0        725  145000   36250     Ghana      Southeast    January   2019  
5        798  119700   55860     Ghana  northcentral        June   2019  
8        700  350000  175000     Benin     southsouth  September   2019  
11       902  180400   45100   Nigeria  northcentral    December   2019  
15       709  354500  177250     Ghana      northwest      April   2019

In [31]:
#total profit made in the month of january

In [32]:
jan_profit = yr_2019[yr_2019['MONTHS'] == 'January'].groupby(['MONTHS'])['PROFIT'].sum()
jan_profit.loc['January']

3263160

In [33]:
#therefore the precentage profit for the months in 2019

In [34]:
month = yr_2019.groupby(['MONTHS'])['PROFIT'].sum()
month/jan_profit.loc['January'] 

MONTHS
April        0.873837
August       0.914083
December     0.627852
February     0.418882
January      1.000000
July         0.902604
June         0.817943
March        0.775512
May          0.788512
November     0.819944
October      0.680589
September    0.579990
Name: PROFIT, dtype: float64

In [35]:
#/*Which particular brand generated the highest profit in Senegal?*/

In [36]:
df_senegal = data[data['COUNTRIES'] =='Senegal']
df_senegal.head()

SALES_ID SALES_REP                    EMAILS       BRANDS  PLANT_COST  \
4      10105   Andrews            andy@gmail.com         hero         150   
9      10110    Howard  howard_freeman@yahoo.com  castle lite         180   
14     10115    Morgan        morganny@gmail.com       trophy         150   
19     10120    Parent           parentty@uk.com    beta malt          80   
24     10125   Jardine            jard@gmail.com  eagle lager         170   

    UNIT_PRICE  QUANTITY    COST  PROFIT COUNTRIES     REGION    MONTHS  YEARS  
4          200       836  167200   41800   Senegal  northeast       May   2017  
9          450       745  335250  201150   Senegal   northwest  October   2017  
14         200       939  187800   46950   Senegal  southsouth    March   2018  
19         150       731  109650   51170   Senegal        west   August   2017  
24         250       715  178750   57200   Senegal   Southeast  January   2018

In [37]:
df_senegal.groupby(['BRANDS'])['PROFIT'].sum().sort_values(ascending = False).head(1)

BRANDS
castle lite    7012980
Name: PROFIT, dtype: int64

The brand that generated the most profit in senegal is Castle lite beer, with a total profit of 7012980

In [38]:
#/*BRAND ANALYSIS*/
#1./*Within the last two years, the brand manager wants to know 
#the top three brands consumed in the francophone countries*/

In [39]:
def check_year(YEARS):
    if (YEARS==2018) or (YEARS==2019):
        return True
    else:
        return False

In [40]:
brand = data[data['YEARS'].apply(check_year)]
brand.head()

SALES_ID SALES_REP               EMAILS       BRANDS  PLANT_COST  \
0     10101   Jardine       jard@gmail.com       trophy         150   
1     10102      Gill      gillhell@uk.com    budweiser         250   
2     10103   Sorvino  sorvi2000@gmail.com  castle lite         180   
3     10104     Jones    jone.ai@yahoo.com  eagle lager         170   
5     10106   Jardine       jard@gmail.com    beta malt          80   

   UNIT_PRICE  QUANTITY    COST  PROFIT COUNTRIES        REGION     MONTHS  \
0         200       725  145000   36250     Ghana      Southeast   January   
1         500       815  407500  203750   Nigeria           west  February   
2         450       937  421650  252990      Togo     southsouth     March   
3         250       765  191250   61200     Benin      northwest     April   
5         150       798  119700   55860     Ghana  northcentral       June   

   YEARS  
0   2019  
1   2018  
2   2018  
3   2018  
5   2019

In [41]:
brand[brand['COUNTRIES'].apply(check_francophone)].groupby(['BRANDS'])['QUANTITY'].sum().sort_values(ascending =False).head(3)

BRANDS
trophy         52899
hero           50846
eagle lager    50630
Name: QUANTITY, dtype: int64

The top 3 brands consumed in the the francophone countries are: (Trophy Hero and Eagle lager)

In [42]:
#2./*Find out the top two choice of consumer brands in Ghana*

In [43]:
data[data['COUNTRIES']=='Ghana'].groupby(['BRANDS'])['QUANTITY'].sum().sort_values(ascending = False).head(2)

BRANDS
eagle lager    25829
castle lite    25806
Name: QUANTITY, dtype: int64

The top 2 consumer brands in Ghana are: Eagle lager and Castle lager beer.

In [44]:
#Find out the details of beers consumed in the past three years in the most oil reach country in West Africa

In [45]:
def check_beer(BRANDS):
    if (BRANDS=='budweiser') or (BRANDS=='hero') or (BRANDS=='trophy') or (BRANDS=='castle lite') or (BRANDS=='eagle lager'):
        return True
    else:
        return False

In [46]:
nig = data[data['COUNTRIES'] =='Nigeria']

nig[nig['BRANDS'].apply(check_beer)].groupby(['BRANDS'])['QUANTITY'].sum().sort_values(ascending=False)

BRANDS
budweiser      26153
eagle lager    25872
hero           25811
trophy         25743
castle lite    25681
Name: QUANTITY, dtype: int64

Nigeria is the reach country in west Africa, with Budweiser being the most consumed beer in the country with a total quantity of 26,153, followed by Eagle lager = 25,872, Hero = 25,811, Trophy =25,743 and Castle lite =25,682

In [47]:
#Favorites malt brand in Anglophone region between 2018 and 2019

In [48]:
def check_malt(BRANDS):
    if (BRANDS =='beta malt') or (BRANDS =='grand malt'):
        return True
    else:
        return False

In [49]:
brand_anglophone = data[(data['COUNTRIES'].apply(check_anglophone)) & (data['YEARS'].apply(check_year)) & (data['BRANDS'].apply(check_malt))]
brand_anglophone.head()
                                            

SALES_ID SALES_REP             EMAILS      BRANDS  PLANT_COST  UNIT_PRICE  \
5      10106   Jardine     jard@gmail.com   beta malt          80         150   
40     10141   Andrews     andy@gmail.com   beta malt          80         150   
61     10162      Gill    gillhell@uk.com   beta malt          80         150   
90     10191   Jardine     jard@gmail.com  grand malt          90         150   
96     10197     Jones  jone.ai@yahoo.com   beta malt          80         150   

    QUANTITY    COST  PROFIT COUNTRIES        REGION     MONTHS  YEARS  
5        798  119700   55860     Ghana  northcentral       June   2019  
40       894  134100   62580     Ghana     northeast        May   2019  
61       931  139650   65170   Nigeria           west  February   2018  
90       902  135300   54120     Ghana      Southeast      July   2019  
96       896  134400   62720   Nigeria      Southeast   January   2019

In [50]:
brand_anglophone.pivot_table(index = ['BRANDS'], values=['QUANTITY'], aggfunc=(np.sum))

QUANTITY
BRANDS              
beta malt      32118
grand malt     33221

The favourite malt drink in the English speaking region(anglophone) is Grand malt

In [51]:
#Which brands sold the highest in 2019 in Nigeria

In [52]:
yr_2019[yr_2019['COUNTRIES'] =='Nigeria'].groupby(['BRANDS'])['PROFIT'].sum().sort_values(ascending = False)

BRANDS
budweiser      1372500
castle lite     914760
eagle lager     672080
beta malt       587230
grand malt      487500
hero            481100
trophy          290150
Name: PROFIT, dtype: int64

The brand in Nigeria with the most sales in 2019 is Budweiser, with a total profit of 1,372,500 

In [53]:
#Favorites brand in South_South region in Nigeria

In [54]:
nig[nig['REGION ']=='southsouth'].groupby(['BRANDS'])['QUANTITY'].sum().sort_values(ascending =False)

BRANDS
eagle lager    4551
trophy         4468
hero           4456
budweiser      4328
castle lite    4287
beta malt      4257
grand malt     3914
Name: QUANTITY, dtype: int64

The favorite brand in the south south region of Nigeria is Eagle lager

In [55]:
#Beer consumption in Nigeria

In [56]:
nig[nig['BRANDS'].apply(check_beer)].groupby(['BRANDS'])['QUANTITY'].sum().sort_values(ascending =False)

BRANDS
budweiser      26153
eagle lager    25872
hero           25811
trophy         25743
castle lite    25681
Name: QUANTITY, dtype: int64

Budweiser is the most consumed beer brand in Nigeria

In [57]:
#Level of consumption of Budweiser in the regions in Nigeria

In [58]:
nig[nig['BRANDS'] =='budweiser'].groupby(['REGION '])['QUANTITY'].sum().sort_values(ascending =False)

REGION 
west             4620
northcentral     4498
southsouth       4328
northeast        4320
northwest        4274
Southeast        4113
Name: QUANTITY, dtype: int64

The western region in Nigeria have the highest consumption rate of Budweiser in the country